db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/TernaryOperatorNode.java

```diff
@@ -885,6 +906,80 @@ public ValueNode substrBind()
 		return this;
 	}
 
+	public ValueNode leftBind() throws StandardException
+	{
+		TypeId	receiverType;
+		TypeId	resultType = TypeId.getBuiltInTypeId(Types.VARCHAR);
+
+		if (receiver.requiresTypeFromContext())
+		{
+			receiver.setType(getVarcharDescriptor());
+			receiver.setCollationUsingCompilationSchema();
+		}
+		if (leftOperand.requiresTypeFromContext())
+		{
+			leftOperand.setType(new DataTypeDescriptor(TypeId.INTEGER_ID, false));
+		}
+		if ((rightOperand != null) && rightOperand.requiresTypeFromContext())
+		{
+			rightOperand.setType(new DataTypeDescriptor(TypeId.getBuiltInTypeId(Types.VARCHAR), true));
+		}
+
+		bindToBuiltIn();
+
+		if (!leftOperand.getTypeId().isIntegerNumericTypeId())
+			throw StandardException.newException(SQLState.LANG_DB2_FUNCTION_INCOMPATIBLE, "LEFT", "FUNCTION");
+
+		receiverType = receiver.getTypeId();
+		switch (receiverType.getJDBCTypeId())
+		{
+			case Types.CHAR:
+			case Types.VARCHAR:
+			case Types.LONGVARCHAR:
+			case Types.CLOB:
+				break;
+			default:
+			{
+				throwBadType("LEFT", receiverType.getSQLTypeName());
+			}
+		}
+
+		if (rightOperand != null)
+		{
+			switch (rightOperand.getTypeId().getJDBCTypeId())
+			{
+				case Types.CHAR:
+				case Types.VARCHAR:
+				case Types.LONGVARCHAR:
+				case Types.CLOB:
+					break;
+				default:
+				{
+					StandardException.newException(SQLState.LANG_DB2_FUNCTION_INCOMPATIBLE, "LEFT", "FUNCTION");
```

Comment:
Why are you using `StandardException.newException` while `throwBadType("LEFT", receiverType.getSQLTypeName());` above?

Reply:
Changed to throwBadType().